ironfish/src/wallet/wallet.ts

```diff
@@ -1584,6 +1584,19 @@ export class Wallet {
       throw new Error(`No account found with name ${account.name}`)
     }
   }
+
+  private async chainHasBlock(hash: Buffer): Promise<boolean> {
+    try {
+      await this.nodeClient.chain.getBlock({ hash: hash.toString('hex') })
+      return true
+    } catch (error) {
+      if (error instanceof Error) {
+        this.logger.error(error.message)
+      }
+
+      return false
```

Comment:
Should we return false here? If this errors out (one remote node for example) it will be as if the block doesn't exist in the chain. Particularly bad for the condition on `start` where if this returns false we reset the account